### ISSUE TITLE
Memory Leak still in Project Red as of v4.5.5.56

### ISSUE THREAD

**Issue by Vaygrim** [MrTJP/ProjectRed#638]
Captured Data: https://ace5852.com/prbug/

The above data was captured using a performance visualizer program on our "Vaygrim's Chance" mod pack test server. We have noticed a memory leak that was first noticed back in v1.03 of my mod pack, which used Project Red v4.5.1.51 and continues now with the current version. I suspect it stems from the routing tables function of the Project Red piping system, unfortunately.

In the above data, click on "Details >>" towards the bottom of the page. This will take you to https://ace5852.com/prbug/pages/18.html which goes through a list of all things currently occupying memory that is being used by the server.

This server has 4GB of memory devoted to it, and usually only has 2 or 3 concurrent players on it.. on average. If you scroll down to the bottom of that page of data you'll notice the following line:

> mrtjp.projectred.transportation.Router 
> First 10 of 34,848 objects 34,848  2,230,272   3,726,632,456

Basically.. this one single item from Project Red is taking up 3.7 of the 4 GB of server memory. The problem starts off small but progressively grows over the course of an hour, forced the server into a crash. We have discovered that this issue can be mitigated using better Java Args however.. it only delays the problem by a few hours. It does not RESOLVE the problem.

Can you please look into this, MrTJP? I would greatly appreciate it. If you need access to our Test Server for the sake of performance evaluating, you are definitely welcome on it. Just let me know. Also you can always find me in Esper.net IRC under the channel #Vaygrim

**Comment by MrTJP** [comment]
I am closing this issue, as there isn't any concrete evidence as to how this memory leak occurred, where it is, or how to solve it. I have yet to run into this issue.
